File: array/0888_fair_candy_swap/0888_fair_candy_swap.go
package main

import "fmt"

func fairCandySwap(A []int, B []int) []int {
	var diff int
	for _, num := range A {
		diff += num
	}
	for _, num := range B {
		diff -= num
	}
	diff /= 2
	data := make(map[int]struct{}, len(A))
	for _, num := range A {
		data[num] = struct{}{}
	}
	for _, num := range B {
		if _, ok := data[num+diff]; ok {
			return []int{num + diff, num}
		}
	}
	return nil
}

func main() {
	A := []int{1, 1}
	B := []int{2, 2}
	res := fairCandySwap(A, B)
	fmt.Println(res)
}

File: array/0888_fair_candy_swap/0888_fair_candy_swap.py
from typing import List


class Solution:
    def fairCandySwap(self, A: List[int], B: List[int]) -> List[int]:
        Sa, Sb = sum(A), sum(B)
        setB = set(B)
        for x in A:
            if x + (Sb - Sa) / 2 in setB:
                return [x, x+(Sb-Sa) / 2]


A = [1,1]
B = [2,2]
res = Solution().fairCandySwap(A, B)
print(res)